src/openms_gui/source/VISUAL/Plot2DCanvas.cpp

```diff
@@ -1160,10 +1160,11 @@ namespace OpenMS
         intensity_max = max(intensity_max, i->getIntensity());
         
         // binning for m/z
-        mzint[int(i->getMZ() * mult)] += i->getIntensity();
-        // ... to later obtain an average m/z value
-        mzcount[int(i->getMZ() * mult)]++;
-        mzsum[int(i->getMZ() * mult)] += i->getMZ();
+        auto intensity = i->getIntensity();
+        mzint[int(i->getMZ() * mult)] += intensity;
+        // ... to later obtain an intensity weighted average m/z value
+        mzcount[int(i->getMZ() * mult)] += intensity;
```

Comment:
mzint and mzcount have the same value, can we get rid of mzcount?

Reply:
yes  I think so.line 1192 needs then to be changed